apollo-federation/src/schema/mod.rs

```diff
@@ -593,6 +596,66 @@ impl FederationSchema {
         Ok(applications)
     }
 
+    pub(crate) fn list_size_directive_applications(
+        &self,
+    ) -> FallibleDirectiveIterator<ListSizeDirective> {
+        let Some(list_size_directive_name) = CostSpecDefinition::list_size_directive_name(self)?
+        else {
+            return Ok(Vec::new());
+        };
+        let Ok(list_size_directive_referencers) = self
+            .referencers()
+            .get_directive(list_size_directive_name.as_str())
+        else {
+            return Ok(Vec::new());
+        };
+
+        let mut applications = Vec::new();
+        for field_definition_position in &list_size_directive_referencers.object_fields {
```

Comment:
You can now use `object_and_interface_fields()` method (I just merged the PR with it).